nx/lib/nx/defn/expr.ex

```diff
@@ -1394,6 +1394,11 @@ defmodule Nx.Defn.Expr do
 
   ## Constant helpers and related optimizations
 
+  defp constant(%{vectorized_axes: [_ | _]} = out, number) do
+    out = %{out | names: Enum.map(out.names, fn _ -> nil end)}
```

Comment:
I don't think this part should be done here, we should preserve the names. Sorry for the confusion.